main.go

```diff
@@ -41,8 +45,10 @@ func main() {
 	config.SsoSecretPtr = flag.String("sso-secret", "", "SSO secret for origin")
 	config.SsoUriPtr = flag.String("sso-url", "", "SSO endpoint eg: http://discourse.forum.com")
 	config.AllowAllPtr = flag.Bool("allow-all", false, "allow all discourse users (default: admin users only)")
+	config.AllowedGroupsPtr = flag.String("allowed-groups", "", "if set, will only auth if the user is member of at least one specified group. eg: 'admins,build'")
```

Comment:
Do we have a need for this feature at present?  The use case I'm aware of that's driving this change is for the list of groups to be exposed to the backend application (via the groups header you've also added), and for the authorization decision to be made by the backend application, not by the auth-proxy.

Reply:
When we turn this on, we will disable the admin requirement, so every user (that has a valid account in the source Discourse) will be able to auth.

Most applications, like Jenkins, will interpret the groups header and apply the correct permissions accordingly. So yeah we don't need this.

What I'm afraid is that some applications won't have this feature, and will let the user go through. So this would be a fail safe feature, so we restrict the auth to only a few specific groups reducing the surface of abuse.

I can break this into two PRs and only merge the groups header thing for now.

Reply:
If you don't want to let everyone auth, don't turn on the "let everyone auth" feature.  I'm not aware of any use case we have now, or planned, which requires auth-proxy to do the authorization decision based on anything other than "is an admin", so in the spirit of YAGNI, let's leave it out until we *do* need it.